### DISCUSSION TITLE
How to persist the column filter object when column filter value is empty

### DISCUSSION THREAD

**Discussion by rohmanhm** [KevinVandy/material-react-table#123] [Q&A]
I've put a state for the `columnFilters` to be like
```ts
const columnFilters = [{ id: 'start', value: 123}]
```
But after I deleted all the value in the text input, it's not only removing the value in `columnFilters`, but also the whole column filter object.
<img width="209" alt="Rohmanshot 2022-09-26 at 15 37 27@2x" src="https://user-images.githubusercontent.com/7524911/192231852-3e576f56-6f8d-4d77-adfb-e3f5e82183f7.png">

So, intead of having an object like this
```ts
const columnFilters = [{ id: 'start', value: null }]
```
I got something like this
```ts
const columnFilters = []
```

Anybody knows how can I achieve that?

**Comment by KevinVandy** [comment]
I think this was intended behavior in TanStack Table itself. What problems does it cause?

**Reply by rohmanhm** [reply]
When the column filter is shown it will trigger the `onColumnFiltersChange` event. In my case, it will also call `setColumnFilters` when nothing has changed.
